### PR TITLE
Added Wasserstein Loss, the Most Popular Loss Function for GANs

### PR DESCRIPTION
**Pull request by hi-sushanta**
Introducing the Wasserstein loss function - the crown jewel of GANs! This state-of-the-art loss function helps both the generator and discriminator reach new heights of performance. The results? Stunningly realistic generated images that even the most discerning eye would mistake for the real thing. It's like magic, but with math! 🎩

**Here is my code!**
Screenshot (10)

## Checklist

- [x] Did you add a function?
- [x] Did you add the tests?
- [x] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

### CODE REVIEW

**Comment by hi-sushanta** [comment]
Does anyone see my changes?

**Comment by hi-sushanta** [comment]
I believe that the community around this repository is no longer active, so I have decided to delete it.

**Comment by niixxaaa** [comment]
Hi, you need to add a hypothesis test in this file `ivy/ivy_tests/test_ivy/test_functional/test_nn/test_losses.py` for the function you've created. 
Thanks :)

**Review by Ishticode — CHANGES_REQUESTED** [review]
Hi @hi-sushanta 
THank you very much for the effort put into this PR. Apologies for lack of correspondence you have received here. 
The two tests that you have added seem to be failing as maybe seen in the logs here https://github.com/unifyai/ivy/actions/runs/8130590086/job/22219132685?pr=27207
under combined test results. For example, test dic wl has a simple failure such as AttributeError: 'list' object has no attribute 'is_integer'. If you are able to fix these issues, please request my review and I should be able to come back to you quickly. Thank you very much for your patience 🙂

File: ivy_tests/test_ivy/test_functional/test_nn/test_losses.py [hunk 1]
```diff
@@ -150,6 +150,78 @@ def test_cross_entropy(
...
+    ),
+)
+def test_disc_wl(
+    dtype_and_p_real, dtype_and_p_fake, test_flags, backend_fw, fn_name, on_device
+):
+    dtype_p_real, p_real = dtype_and_p_real
+    dtype_p_fake, p_fake = dtype_and_p_fake
+
+    helpers.test_function(
+        input_dtypes=[dtype_p_real, dtype_p_fake],
```
Comment (Ishticode) [inline ivy_tests/test_ivy/test_functional/test_nn/test_losses.py:184]: you should try this as `dtype_p_real + dtype_p_fake`

**Review by Ishticode — CHANGES_REQUESTED** [review]
it was in the ci as seen here
image
I think it is due to the way the input dtypes are put in the tests as pointed for test_disc_wl below.

**Comment by Ishticode** [comment]
HI. As you can see the is_integer issue is not directly in your code but it comes from how your test expected dtypes to be passed. It should not be a list of lists. You should be able to run tests locally and see these issues easily. Thanks

**Review by Sam-Armstrong — APPROVED** [review]

**Pull request merged**